lib/queue.js

```diff
@@ -570,7 +570,11 @@ async function redisClientDisconnect(client) {
 }
 
 Queue.prototype.disconnect = async function() {
-  await Promise.all(this.clients.map(redisClientDisconnect));
+  await Promise.all(
+    this.clients.map(client =>
+      client.blocked ? client.disconnect() : redisClientDisconnect(client)
```

Comment:
Not sure if that will [fix the issue][1]. I'll do experiment and see.

BTW, I guess this logic should move to `redisClientDisconnect()` itself, so that if that is called from anywhere else, then the logic still applies. I'm not 10% sure though whether that'd be a good idea.. 🤔 

[1]: https://github.com/OptimalBits/bull/issues/2372

Reply:
It is the opposite actually, it must only be used in this particular case.